error.go

```diff
@@ -397,3 +397,46 @@ func Append(left error, right error) error {
 	errors := [2]error{left, right}
 	return fromSlice(errors[0:])
 }
+
+// AppendInto appends an error into the destination of an error pointer and
+// returns whether the error being appended was non-nil.
+//
+// 	var err error
+// 	multierr.AppendInto(&err, r.Close())
+// 	multierr.AppendInto(&err, w.Close())
+//
+// The above is equivalent to,
+//
+// 	err := multierr.Append(r.Close(), w.Close())
+//
+// As AppendInto reports whether the provided error was non-nil, it may be
+// used to build a multierr error in a loop more ergonomically.
+//
+// 	var err error
+// 	for line := range lines {
+// 		var item Item
+// 		if multierr.AppendInto(&err, parse(line, &item)) {
+// 			continue
+// 		}
+// 		items = append(items, item)
+// 	}
+//
+// Compare the loop body above with a version that relies solely on Append.
+//
+// 	var item Item
+// 	if parseErr := parse(line, &item); parseErr != nil {
+// 		err = multierr.Append(err, parseErr)
+// 		continue
+// 	}
+// 	items = append(items, item)
+func AppendInto(into *error, err error) (errored bool) {
+	if err == nil {
+		return false
+	}
+	// We will panic if 'into' is nil. This is not documented above
```

Comment:
Yes, but what's the failure look like? Would it still be better to add an explicit nil guard here, so that we can write a  better panic message?

Reply:
Fair. How about a panic with the message "'into' must not be a nil pointer"?